components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/JWTTokenIssuer.java

```diff
@@ -899,6 +887,10 @@ private JWTClaimsSet handleCustomClaims(JWTClaimsSet.Builder jwtClaimsSetBuilder
                                               OAuthTokenReqMessageContext tokenReqMessageContext, OAuthAppDO oAuthAppDO)
             throws IdentityOAuth2Exception {
 
+        if (tokenReqMessageContext != null && tokenReqMessageContext.isPreIssueAccessTokenActionsExecuted()) {
+            return handleCustomClaimsInPreIssueAccessTokenResponse(jwtClaimsSetBuilder, tokenReqMessageContext);
```

Comment:
retuning here will not invoke `claimsCallBackHandler.handleCustomClaims(jwtClaimsSetBuilder, tokenReqMessageContext)`. This cloud be a issue that we handle user attributes for access token there. Please check

Reply:
ack

Reply:
Before executing the action, claims were added to the token messsage context in the `PreIssueAccessTokenRequestBuilder` class. Hence returning after here won't be a problem.

Also modified the logic in `PreIssueAccessTokenRequestBuilder` class to get the claimhandler using a util class after the introduction of new `JWTAccessTokenOIDCClaimsHandler` class by https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2552/commits/538d658a318144142b905011f5834874384b4466

[1] - https://github.com/wso2-extensions/identity-inbound-auth-oauth/blob/548cf3916ba720d7369407bae2598e260acf9738/components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/action/PreIssueAccessTokenRequestBuilder.java#L284C13-L288C1
[2] - https://github.com/wso2-extensions/identity-inbound-auth-oauth/blob/master/components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/JWTAccessTokenOIDCClaimsHandler.java